modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/display/context/ViewClientExtensionEntryPartDisplayContext.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.client.extension.web.internal.display.context;
+
+import com.liferay.client.extension.type.CET;
+import com.liferay.client.extension.type.annotation.CETProperty;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.util.CamelCaseUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.lang.reflect.Method;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import javax.portlet.PortletRequest;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public class ViewClientExtensionEntryPartDisplayContext<T extends CET> {
+
+	public ViewClientExtensionEntryPartDisplayContext(
+		T cet, PortletRequest portletRequest) {
+
+		_cet = cet;
+		_portletRequest = portletRequest;
+	}
+
+	public T getCET() {
+		return _cet;
+	}
+
+	public String getLabel(Method method) {
+		CETProperty cetProperty = method.getAnnotation(CETProperty.class);
+
+		String label = cetProperty.label();
+
+		if (Validator.isBlank(label)) {
+			label = CamelCaseUtil.fromCamelCase(cetProperty.name());
+		}
+
+		return LanguageUtil.get(_getHttpServletRequest(), label);
+	}
+
+	public Collection<Method> getPropertyMethods() {
+		Class<? extends CET> clazz = _cet.getClass();
+
+		List<Method> methods = new ArrayList<>();
+
+		for (Class<?> iface : clazz.getInterfaces()) {
```

Comment:
Really like this approach to grab any annotated method not present in sub-interfaces :)